src/msg/async/Event.cc

```diff
@@ -361,21 +362,17 @@ int EventCenter::process_events(int timeout_microseconds)
 
   if (external_num_events.load()) {
     external_lock.lock();
-    if (external_events.empty()) {
-      external_lock.unlock();
-    } else {
-      deque<EventCallbackRef> cur_process;
-      cur_process.swap(external_events);
-      external_num_events.store(0);
-      external_lock.unlock();
-      while (!cur_process.empty()) {
-        EventCallbackRef e = cur_process.front();
-        ldout(cct, 20) << __func__ << " do " << e << dendl;
-        if (e)
-          e->do_request(0);
-        cur_process.pop_front();
-        numevents++;
-      }
+    deque<EventCallbackRef> cur_process;
+    cur_process.swap(external_events);
+    external_num_events.store(0);
+    external_lock.unlock();
+    while (!cur_process.empty()) {
+      EventCallbackRef e = cur_process.front();
+      ldout(cct, 20) << __func__ << " do " << e << dendl;
+      if (e)
```

Comment:
under what condition, could we call `EventCenter::dispatch_event_external()` with a  nullptr?